src/imitation/algorithms/density.py

```diff
@@ -343,7 +343,7 @@ def __call__(
         rew_array = np.asarray(rew_list, dtype="float32")
         return rew_array
 
-    def train_policy(self, n_timesteps: int = int(1e6), **kwargs):
+    def train_policy(self, n_timesteps: int = int(1e6), **kwargs: Any) -> None:
```

Comment:
We could try to be more specific than this, but ParamSpec is not (yet) a very well supported feature by most type checkers, and it might not be worth the effort of typing it.

Reply:
I've not used `ParamSpec` much so may be missing something, but how would it help us here? It seems designed in cases where the `P = ParamSpec("P")` variable is reused, like decorators or other higher-order functions. But if we're returning `None` I don't think it makes a difference.

Reply:
I meant for Any, not for None haha. The problem with this is that any subtype has to take any kwarg in the signature, but you only really want it to take the kwargs that `.learn()` can take. But not sure how you can use ParamSpec for this now I think about it. I think it was designed as a generic that is filled in implicitly at the callsite, not for extracting the type signature of an explicit method (which tbh should be a valid use case).